File: src/java/src/main/java/org/elastos/hive/vault/scripting/RunScriptParams.java
package org.elastos.hive.vault.scripting;

import com.google.gson.annotations.SerializedName;

class RunScriptParams {
	@SerializedName("context")
	private Context context;
	@SerializedName("params")
	private Object params;

	public RunScriptParams setContext(Context context) {
		this.context = context;
		return this;
	}

	public RunScriptParams setParams(Object params) {
		this.params = params;
		return this;
	}
}

File: src/restclient/scripting/runscriptparams.ts
import { Context } from "./context";

export class RunScriptParams {
	private context: Context;
	private params: Object;

	setContext( context: Context): RunScriptParams {
		this.context = context;
		return this;
	}

	setParams( params: object) : RunScriptParams {
		this.params = params;
		return this;
    }
    
    toString() : string {
        return JSON.stringify(this);
    }
}
